### ISSUE TITLE
Can't use aot with a let var in custom scripts

### ISSUE THREAD

**Issue by Jonathan002** [angular/angular-cli#8561]
### Versions

```
@angular/cdk: 5.0.0-rc0
@angular/cli: 1.5.2
@angular/material: 5.0.0-rc0
@angular/platform-server: 4.4.6
@angular/tsc-wrapped: 4.4.6
@angular-devkit/build-optimizer: 0.0.33
@angular-devkit/core: 0.0.20
@angular-devkit/schematics: 0.0.35
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.8.2
@schematics/angular: 0.1.2
typescript: 2.4.2
webpack: 3.8.1
```

### Repro steps

* Create New Angular CLI Project
* Create a script file in src and add it to angular.cli
* Create a let var in custom script -   
      /src/custom-script.js - `let object = {}`
* Try to compile the project with --aot and --prod flag. `npm run build --aot --prod`

### Observed behavior
```
<!-- Normally this includes a stack trace and some more information. -->
```
You will get the error:
```
(node:22021) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
and no dist folder will be created.

### Desired behavior

Successful aot build compilation and creation of a dist folder for the project.

### Mention any other details that might be useful (optional)
I've created a repository with the custom script problem setup in .angular-cli.json.

Link: https://github.com/Jonathan002/angular-cli-broken-aot-demo

**Comment by filipesilva** [comment]
Heya @Jonathan002, thanks for opening this and including a repro. I think I know why it happens: we Uglify  scripts with an older version of Uglify that doesn't support ES6 constructs such as let.

**Comment by tomastrajan** [comment]
I'll do it, if I understand correctly it's about creating PR with version bump of `uglify-js` for `webpack-concat-plugin` repo and then owner has to release it and then we bump version used by `@angular/cli` right?

**Comment by tomastrajan** [comment]
Ok so I created a  [PR](https://github.com/hxlniada/webpack-concat-plugin/pull/26) in the [webpack-concat-plugin](https://github.com/hxlniada/webpack-concat-plugin) which solves the issue when testing against the provided reproduction repo. Also tried to run Angular CLI tests with the updated plugin locally and all worked fine.

**Comment by filipesilva** [comment]
@tomastrajan thanks for taking this on. I also pinged the library owner there about this change.

**Comment by tomastrajan** [comment]
Ok so the `uglify-es` `2.0.0` was released! I'll create a PR with the version bump.

**Comment by tomastrajan** [comment]
Ok, so i created the PR, hope it's all good @filipesilva .

**Comment by tomastrajan** [comment]
This seems to be solved by [PR#8849](https://github.com/angular/angular-cli/pull/8849) .

**Comment by clydin** [comment]
Closing as fixed by #8849
